app/router/routes.py

```diff
@@ -0,0 +1,26 @@
+import sys
+import os
+
+sys.path.insert(
+    0, os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+)  # Fixes import errors
+
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel
+from api.url_generator import get_api_url
+
+
+router = APIRouter()
+
+
+class QueryRequest(BaseModel):
+    query: str
+
+
+@router.post("/generate_url/")
+async def generate_url(request: QueryRequest):
+    try:
+        api_url = get_api_url(request.query)
+        return {"response": api_url}
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
```

Comment:
```suggestion
        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e))
```
I'm assuming the error `e` here will be coming from the `get_api_url` where we have informative messages for errors.
Would be a good idea to write a test for this.